download/upgradeRecipe/upgradeRecipe8.adoc

```diff
@@ -44,33 +42,79 @@ Any backwards incompatible changes are annotated with a [.label.label-danger.lab
 //We currently intend to support a minimal version of Java 11 throughout the entire 8.x series.
 
 
+[.upgrade-recipe-minor.upgrade-recipe-public-api]
+=== Refrain from using Java serialization
+
+Java serialization is
+https://cr.openjdk.java.net/~briangoetz/amber/serialization.html[considered by its authors to be poorly designed] and it was suggested that the capability would, at some point in time, be removed from the platform altogether.
+In OptaPlanner 8, we have removed most uses of the `Serializable` marker interface from the public API and we encourage users to
+https://docs.optaplanner.org/latest/optaplanner-docs/html_single/index.html#integrationWithPersistentStorage[serialize
+via JAXB or JSON].
+
+[.upgrade-recipe-major.upgrade-recipe-public-api]
+=== Refrain from using `ScoreDirectorFactory`
```

Comment:
I'd change to something like this (or shorter):
`SolverFactory.getScoreDirectorFactory()` replaced by `ScoreManager`